### PR TITLE
Add configuration knob for dataplane readiness timeout.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Allows for workaround if liveness check is spuriously failing.

Increase default from 20s to 90s.  We see many more false positives than
genuine hangs.

## Related issues/PRs

This would have been useful to work around the issues in #4118, for example.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Add DataplaneWatchdogTimeout configuration parameter to Felix to allow the watchdog timeout to be increased.  Prevents spurious liveness failures on heavily loaded systems or systems operating at very high pod scale.  Increase default timeout to 90s (from 20s) after seeing too many issues where the liveness check failed spuriously.
```

### CODE REVIEW

File: felix/fv/health_test.go
Comment (neiljerram) [inline felix/fv/health_test.go:225]: It would be nice to add a test for when the timeout is set to a larger value.  (Or is the new default of 90s.)

File: api/pkg/apis/projectcalico/v3/felixconfig.go
Comment (neiljerram) [inline api/pkg/apis/projectcalico/v3/felixconfig.go:67]: Should this be commented?  (Obviously most config fields are.  But perhaps this initial group is special in some way?)

**Review by neiljerram — COMMENTED** [review]
Couple of queries.

File: api/pkg/apis/projectcalico/v3/felixconfig.go
Comment (robbrockbank) [inline api/pkg/apis/projectcalico/v3/felixconfig.go:68]: useed -> used

File: calico/reference/resources/felixconfig.md
Comment (robbrockbank) [inline calico/reference/resources/felixconfig.md:39]: I think this list is (mostly) in alphanumeric order.

File: felix/dataplane/linux/int_dataplane.go
Comment (neiljerram) [inline felix/dataplane/linux/int_dataplane.go:843]: nit: Would be nice to log the too-low value, in case user is confused about what they've configured.

**Review by neiljerram — APPROVED** [review]
LGTM, with one nit, and the typo that Rob noted.

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, squash the commits when I merge it, and leave the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, squash the commits when I merge it, and leave the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, squash the commits when I merge it, and leave the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, squash the commits when I merge it, and leave the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, squash the commits when I merge it, and leave the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, squash the commits when I merge it, and leave the branch after I've merged it.

**Pull request merged**
